rotkehlchen/exchanges/bitstamp.py

```diff
@@ -0,0 +1,711 @@
+            self,
+            start_ts: Timestamp,  # pylint: disable=unused-argument
+            end_ts: Timestamp,
+            options: Dict[str, Any],
+            case: str,
+    ) -> Union[List[Trade], List[AssetMovement]]:
+        """Request a Bitstamp API v2 endpoint paginating via an options
+        attribute.
+
+        Currently it targets the "user_transactions" endpoint. For supporting
+        other endpoints some amendments should be done (e.g. request method,
+        loop that processes entities).
+
+        Pagination attribute criteria per endpoint:
+          - user_transactions:
+            * since_id: from <Trade>.id + 1 (if db trade exists), else 1
+            * limit: 1000 (API v2 default using `since_id`).
+            * sort: 'asc'
+        """
+        endpoint: Literal['user_transactions']
+        response_case: Literal['trades', 'asset_movements']
+        if case == 'trades':
+            endpoint = 'user_transactions'
+            raw_result_type_filter = USER_TRANSACTION_TRADE_TYPE
+            response_case = 'trades'  # same than `case`
+            case_pretty = 'trade'
+        elif case == 'asset_movements':
+            endpoint = 'user_transactions'
+            raw_result_type_filter = USER_TRANSACTION_ASSET_MOVEMENT_TYPE
+            response_case = 'asset_movements'  # same than `case`
+            case_pretty = 'asset movement'
+        else:
+            raise AssertionError(f'Unexpected Bitstamp case: {case}.')
+
+        # Check required options per endpoint
+        if endpoint == 'user_transactions':
+            assert options['limit'] == API_MAX_LIMIT
+            assert options['since_id'] >= USER_TRANSACTION_MIN_SINCE_ID
+            assert options['sort'] == USER_TRANSACTION_SORTING_MODE
+            assert options['offset'] == 0
+
+        call_options = options.copy()
+        limit = options.get('limit', API_MAX_LIMIT)
+        results = []
+        while True:
+            response = self._api_query(
+                endpoint=endpoint,
+                method='post',
+                options=call_options,
+            )
+            if response.status_code != HTTPStatus.OK:
+                return self._process_unsuccessful_response(
+                    response=response,
+                    case=response_case,
+                )
+            try:
+                response_list = rlk_jsonloads_list(response.text)
+            except JSONDecodeError:
+                msg = f'Bitstamp returned invalid JSON response: {response.text}.'
+                log.error(msg)
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying Bistamp trades: {msg}',
+                )
+                no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
+                return no_results
+
+            has_results = False
+            is_result_timesamp_gt_end_ts = False
+            for raw_result in response_list:
+                if (
```

Comment:
Skipping by `type` only applies to the "user_transactions" endpoint.
We should be able to use this function for paginating other endpoints at least via limit/offset/datetime, but also any other particular endpoint attribute.

Reply:
I see you already removed it. Good.